core/src/main/java/org/mule/runtime/core/transformer/types/DataTypeFactory.java

```diff
@@ -308,4 +306,52 @@ else if (classInMap != null)
         proxyClassCache.put(typeName, new ProxyIndicator(type, isProxy));
         return isProxy;
     }
+
+    public static <T> DataType<T> createFromDataType(DataType<T> typeFrom, String mimeType, String encoding)
+    {
+        if (typeFrom instanceof CollectionDataType)
+        {
+            Class<? extends Collection<?>> collectionType = (Class<? extends Collection<?>>) typeFrom.getType();
+            Class<?> itemType = ((CollectionDataType) typeFrom).getItemType();
+            if (itemType == null)
+            {
+                return new CollectionDataType(collectionType, Object.class, mimeType, encoding);
+            }
+            else
+            {
+                return new CollectionDataType(collectionType, itemType, mimeType, encoding);
+            }
+        }
+        else
+        {
+            return new SimpleDataType<>(typeFrom.getType(), mimeType, encoding);
+        }
+    }
+
+    public static <T> DataType<T> createFromDataTypeWithMimeType(DataType<T> typeFrom, String mimeType)
```

Comment:
Method name is inconsistent with other methods. I understand it's because you can't overload, but it's not very nice.